docs/ENCRYPTED_STORAGE.md

```diff
@@ -0,0 +1,440 @@
+# Encrypted Storage Implementation
+
+This document describes how Bottlerocket implements encrypted storage for both the data partition (`/local`) and the datastore directory (`/.bottlerocket/datastore`).
+
+## Overview
+
+Bottlerocket's encrypted storage feature provides:
+
+- **Block device encryption** using LUKS2 for the `/local` partition
+- **Directory encryption** using fscrypt for `/.bottlerocket/datastore`
+- **TPM2-based key management** with automatic unlocking
+- **Boot phase measurements** for attestation and policy enforcement
+- **Transparent operation** with no user intervention required
+
+All encryption keys are sealed to TPM2 PCRs, ensuring data can only be decrypted when the system boots in a trusted state.
+
+## Architecture
+
+### Components
+
+1. **rottweiler** - Unified storage encryption helper (Rust binary)
+2. **systemd services** - Orchestrate encryption, unlocking, and measurements
+3. **TPM2** - Hardware security module for key sealing and measurements
+4. **systemd-creds** - Encrypts keys with TPM2 PCR binding
+5. **cryptsetup** - LUKS2 block device encryption
+6. **fscrypt** - Directory-level encryption
+
+### Key Storage
+
+Encryption keys are stored in `/.bottlerocket/keystore/` as TPM2-sealed credentials:
+- `datastore` - Key for `/.bottlerocket/datastore` (fscrypt)
+- `bottlerocket-data` - Key for `/dev/disk/by-partlabel/BOTTLEROCKET-DATA` (LUKS2)
+
+Keys are:
+- 64 bytes of random data from `/dev/random`
+- Encrypted with `systemd-creds` using TPM2 PCR binding
```

Comment:
Is this encryption bound to any PCRs that we shouldn't be touching?

Reply:
This is covered in the "TPM2 PCR Binding" section of the doc.